src/web/toga_web/app.py

```diff
@@ -13,33 +14,246 @@ def __init__(self, interface):
+        #                     aria-controls="navbarsExample07" aria-expanded="false"
+        #                     aria-label="Toggle navigation">
+        #                 <span class="navbar-toggler-icon"></span>
+        #             </button>
+
+        #             <div class="collapse navbar-collapse" id="navbarsExample07">
+        #                 <ul class="navbar-nav mr-auto">
+        #                     <!--li class="nav-item">
+        #                         <a class="nav-link" href="#">Menu</a>
+        #                     </li>
+        #                     <li class="nav-item">
+        #                         <a class="nav-link disabled" href="#">Disabled menu</a>
+        #                     </li-->
+        #                 </ul>
+        #                 <ul class="navbar-nav ml-auto">
+        #                     <li class="nav-item dropdown">
+        #                         <a class="nav-link dropdown-toggle"
+        #                             href="http://example.com" id="dropdown07"
+        #                             data-toggle="dropdown"
+        #                             aria-haspopup="true"
+        #                             aria-expanded="false">Help</a>
+        #                         <div class="dropdown-menu" aria-labelledby="dropdown07">
+        #                             <a class="dropdown-item" href="#">About</a>
+        #                             <a class="dropdown-item" href="#">Preferences</a>
+        #                         </div>
+        #                     </li>
+        #                 </ul>
+        #             </div>
+        #         </div>
+        #     </nav>
+        # """
+
+        # Create the menus.
+        self.create_menus()
+
+        # Call user code to populate the main window
+        self.interface.startup()
+
+    def create_menus(self):
+        self._menu_groups = {}
+        submenu = None
+
+        for cmd in self.interface.commands:
+            if cmd == toga.GROUP_BREAK:
+                submenu = None
+            elif cmd == toga.SECTION_BREAK:
+                # TODO - add a section break
+                pass
+            else:
+                # TODO - this doesn't handle submenus properly;
+                # all menu items will appear in their root group.
+                submenu = self._menu_groups.setdefault(cmd.group, [])
+
+                menu_item = create_element(
+                    "a",
+                    classes=["dropdown-item"] + ([] if cmd.enabled else ['disabled']),
+                    content=cmd.text,
+                )
+                menu_item.onclick = cmd.action
+
+                submenu.append(menu_item)
+
+        menu_container = create_element(
+            "ul",
+            classes=["navbar-nav", "mr-auto"],
+        )
+
+        for group, items in self._menu_groups.items():
+            if group != toga.Group.HELP:
+                submenu = create_element(
```

Comment:
Why is the Help menu being handled separately? If there's a good reason for doing this, the duplicate code should be factored out.

Reply:
The root "Help" menu is very slightly different (because it's right aligned); but there's definitely opportunities for refactoring. I've cleaned this up.